core/templates/dev/head/pages/exploration_player/ImagePreloaderServiceSpec.js

```diff
@@ -386,11 +399,21 @@ describe('Image preloader service', function() {
       $httpBackend.expect('GET', requestUrl4).respond(408);
       expect(ips.getFilenamesOfImageCurrentlyDownloading().length).toBe(3);
       $httpBackend.flush(3);
-      expect(ips.isInFailedDownload('sIOFeedback.png')).toBe(true);
+      expect(ips.isInFailedDownload(
+        'sIOFeedback_height_50_width_50.png')).toBe(true);
       expect(ips.getFilenamesOfImageCurrentlyDownloading().length).toBe(1);
       $httpBackend.flush(1);
-      expect(ips.isInFailedDownload('s6Hint1.png')).toBe(true);
+      expect(ips.isInFailedDownload(
+        's6Hint1_height_60_width_60.png')).toBe(true);
       ips.restartImagePreloader('State 6');
-      expect(ips.isInFailedDownload('s6Hint1.png')).toBe(false);
+      expect(ips.isInFailedDownload(
+        's6Hint1_height_60_width_60.png')).toBe(false);
     });
+
+  it('should calculate the dimensions of the image file', function() {
+    dimensions = ips.getDimensionsOfImage(
+      'sIOFeedback_height_50_width_50.png');
```

Comment:
Have more tests. Different file extensions, invalid filenames (e.g. with no dimensions), invalid filenames where there are chars after the final "width_50", etc.

Reply:
Done